src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessageView.razor

```diff
@@ -298,17 +300,16 @@
             ForwardedPeerAuthor = forwardedPeerAuthor,
         };
         return result.Equals(_rendered) ? _rendered! : result;
-    }
-
-    private async Task<ChatEntry?> GetRepliedEntry(CancellationToken cancellationToken) {
-        if (Entry.RepliedEntryLocalId is not { } repliedEntryLocalId)
-            return null;
 
-        var repliedEntryId = new ChatEntryId(Message.Entry.ChatId, Message.Entry.Kind, repliedEntryLocalId, AssumeValid.Option);
-        var repliedEntry = await Chats.GetEntry(Session, repliedEntryId, cancellationToken)
-            ?? ChatEntry.Removed(repliedEntryId);
+        async Task<ChatEntry?> GetRepliedEntry(CancellationToken cancellationToken1) {
+            if (Entry.RepliedEntryLocalId is not { } repliedEntryLocalId)
+                return null;
 
-        return repliedEntry;
+            var messageEntry = chatMessage.Entry;
+            var repliedEntryId1 = new ChatEntryId(messageEntry.ChatId, messageEntry.Kind, repliedEntryLocalId, AssumeValid.Option);
```

Comment:
Use chatId instead of messageEntry.ChatId, it's better to stored messageEntry.Kind in the begining as well.

Reply:
We have already captured `messageEntry`